src/store/modules/job/actions.ts

```diff
@@ -765,6 +765,125 @@ const actions: ActionTree<JobState, RootState> = {
+          },
+          'statusId': "SERVICE_PENDING",
+          'systemJobEnumId': job.systemJobEnumId
+        } as any
+
+        
+        if (job?.runtimeData?.shopifyConfigId) {
+          const shopifyConfig = this.state.user.shopifyConfigs.find((config: any) => {
+            return config.shopId === shopId;
+          })
+
+          params['shopifyConfigId'] = shopifyConfig.shopifyConfigId;
+          params['jobFields']['shopId'] = shopId;
+        }
+
+        // checking if the runtimeData has productStoreId, and if present then adding it on root level
+        job?.runtimeData?.productStoreId?.length >= 0 && (params['productStoreId'] = payload.eComStoreId)
+        job?.priority && (params['SERVICE_PRIORITY'] = job.priority.toString())
+
+        // assigning '' (empty string) to all the runtimeData properties whose value is "null"
+        job.runtimeData && Object.keys(job.runtimeData).map((key: any) => {
+          if (job.runtimeData[key] === 'null') job.runtimeData[key] = ''
+        })
+        jobParams.push({ ...job.runtimeData, ...params, jobId: job.jobId });
+      })
+    })
+
+    Promise.allSettled(jobParams.map(async (param: any) => {
+      const resp: any = await JobService.scheduleJob(param);
+      if(resp.status === 200 && !hasError(resp)){
+        // Removing the scheduled job
+        dispatch('removeBulkJob', param.jobId);
+        return Promise.resolve(resp);
+      } else {
+        failedJobs++;
+        return Promise.reject(resp);
+      }
+    })).then((resps: any) => {
+      resps.some((resp: any) => {
+        if(resp.status === "rejected") {
+          return showToast(translate("Failed to schedule service(s)", {count: failedJobs}))
+        } else {
+          return showToast(translate('Services have been scheduled'))
+        }
+      })
+    })
+  },
+  setBulkJobData({ commit, state }, payload) {
+    let bulkJobs = JSON.parse(JSON.stringify(state.bulk.jobs));
+    const value = payload.value, type = payload.type;
+    if (payload.global) {
+      commit(types.JOB_BULK_DATA_UPDATED, { type, value });
+      bulkJobs = bulkJobs.map((job: any) => {
+        // handling special case for slow frequency jobs
+        if (type === 'frequency' && job.freqType === 'slow') {
+          showToast(translate("Some jobs have slow frequency type, hence, feasible frequency will be set automatically"))
+          return ["HOURLY", "EVERY_6_HOUR", "EVERYDAY"].includes(value) ? ({...job, [type]: value}) : ({...job, [type]: 'EVERYDAY'})
+        }
+        else return ({ ...job, [type]: (state.bulk as any)[type] })
+      });
+    } else {
+      bulkJobs.forEach((job: any) => { if (job.jobId === payload.jobId) { job[type] = value }});
+    }
+    commit(types.JOB_BULK_UPDATED, bulkJobs);
+  },
+  removeBulkJob({ commit, state }, jobId) {
+    const bulkJobs = JSON.parse(JSON.stringify(state.bulk.jobs)).filter((job: any) => !(job.jobId === jobId));
```

Comment:
Add comment here, why we are filtering jobs here.

```suggestion
    const bulkJobs = JSON.parse(JSON.stringify(state.bulk.jobs)).filter((job: any) => job.jobId !== jobId);
```

Reply:
Updated sir